Library/Homebrew/utils/curl.rb

```diff
@@ -118,36 +118,37 @@ def curl(*args, print_stdout: true, **options)
       result
     end
 
-    def curl_download(*args, to: nil, partial: true, **options)
+    def parse_headers(headers)
+      return {} unless headers
+
+      # Skip status code
+      headers.split("\r\n")[1..].to_h { |h| h.split(": ") }
+    end
+
+    def curl_download(*args, to: nil, try_partial: true, **options)
       destination = Pathname(to)
       destination.dirname.mkpath
 
-      if partial
-        range_stdout = curl_output("--location", "--range", "0-1",
-                                   "--dump-header", "-",
-                                   "--write-out", "%\{http_code}",
-                                   "--output", "/dev/null", *args, **options).stdout
-        headers, _, http_status = range_stdout.partition("\r\n\r\n")
+      if try_partial
+        range_stdout = curl_output("--location", "--head", *args, **options).stdout
+        headers = parse_headers(range_stdout.split("\r\n\r\n").first)
 
-        supports_partial_download = http_status.to_i == 206 # Partial Content
-        if supports_partial_download &&
+        # Any value for `accept-ranges` other than none indicates that the server supports partial requests.
+        # Its absence indicates no support.
+        supports_partial = headers["accept-ranges"] && headers["accept-ranges"] != "none"
```

Comment:
Did you make this change because the default hash value isn't necessarily `nil`?

Reply:
It should always be `nil`, but I'm thinking more about other edge cases.

Using `key?` means that there's no ambiguity about what we're trying to check for here. At the moment it probably doesn't make much of a difference but it's hypothetically possible that `header["accept-ranges"]` could end up being `false` which might incorrectly set `supports_partial`. Again, this specific example is hypothetical because it should always be a string for now, but I think using `key?` is a better long-term solution. We can be sure we're checking exactly what we want to check if that makes sense.

Plus, in my opinion at least, it makes understanding the line a little better because it now reads as "if `headers` has the key `accept-ranges` and the value is not `none`". There's no need to wonder if `headers["accept-ranges"]` is a boolean or anything like that.

Reply:
>It should always be nil, but I'm thinking more about other edge cases.

I suppose in this case it should always be nil, but you can [set the default value for a hash](https://docs.ruby-lang.org/en/2.0.0/Hash.html#method-i-default-3D).

> Using key? means that there's no ambiguity about what we're trying to check for here. At the moment it probably doesn't make much of a difference but it's hypothetically possible that header["accept-ranges"] could end up being false which might incorrectly set supports_partial. Again, this specific example is hypothetical because it should always be a string for now, but I think using key? is a better long-term solution. We can be sure we're checking exactly what we want to check if that makes sense.

Makes sense. Even without potential edge cases, I think it makes the code clearer (as you suggested). Thanks for the edit! :)